ethereumj-core/src/test/java/org/ethereum/util/RLPTest.java

```diff
@@ -1773,4 +1774,28 @@ public void testDepthLimit() {
         Transaction tx = new Transaction(rlpEl3Level3.getRLPData());
         assertArrayEquals(Hex.decode("3e15c330e00f3af7ceeeeb10c27237845a034399c151f8d69708fd3ee45c09c6"), tx.getHash());
     }
+
+    @Test
+    public void testWrapUnwrap() {
+
+        byte[] nullArray    = RLP.encodeElement(new byte[] {});
+        byte[] singleZero   = RLP.encodeElement(new byte[] {0});
+        byte[] singleByte   = RLP.encodeElement(new byte[] {1});
+        byte[] shortItem    = RLP.encodeElement(new byte[] {(byte) 0x81});
+        byte[] longItemData = new byte[57]; new Random().nextBytes(longItemData);
+        byte[] longItem     = RLP.encodeElement(longItemData);
+        byte[] shortList    = RLP.encodeList(shortItem, shortItem, shortItem);
+        byte[] longList     = RLP.encodeList(longItem, longItem, longItem);
+
+        byte[] encoded = RLP.wrapList(nullArray, singleZero, singleByte, shortItem, longItem, shortList, longList);
```

Comment:
I don't understand this
You have `encodeElement` for each data piece inside `wrapList` but prepared everything above with `encodeElement` too
I think you should put raw data instead

Reply:
I mean that you test it on safe already encoded data, while it could be used with unsafe notencoded data.